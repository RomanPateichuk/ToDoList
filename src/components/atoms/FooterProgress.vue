<template>
  <div :class="$style.progress">{{ X }}/{{ getActiveCount }} left</div>
</template>

<script>
import { mapGetters } from "vuex";

export default {
  computed: {
    ...mapGetters(["getActiveCount"]),

    // getTaskActiveCount: function () {
    //   return this.get_tasks_active.reduce(
    //     (prev, el) => (el.checked == false ? prev + 1 : prev),
    //     0
    //   );
    // },
    // getTaskDoneCount: function () {
    //   return this.get_tasks_active.reduce(
    //     (prev, el) => (el.checked == true ? prev + 1 : prev),
    //     0
    //   );
    // },
  },
  methods: {
    // getTaskDoneCount: function(){
    //   return this.get_tasks_active.reduce((prev, el) => (el.checked == true) ? prev + 1: prev, 0);
    // },
  },
};
</script>

<style lang="scss" module>
@import "@/assets/scss/main.scss";
</style>
