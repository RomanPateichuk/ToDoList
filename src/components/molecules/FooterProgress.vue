<template>
  <div :class="$style.progress">1/3 left</div>
</template>

<style lang="scss" module>
@import "@/assets/scss/main.scss";
</style>
