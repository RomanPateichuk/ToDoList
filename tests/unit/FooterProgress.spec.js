import { mount } from "@vue/test-utils";
import FooterProgress from "../../src/components/atoms/FooterProgress.vue";


describe("Тестирование компонента FooterProgress:", () => {
const wrapper = mount(FooterProgress,{
    computed:{
        getDoneCount: ()=> 2,
        getTasks: ()=>5
    }
});

   
test("соответствие снимку", () => {
    expect(wrapper).toMatchSnapshot();
})

test("Проверка работы геттеров", () => {
    expect(wrapper.find("div").text()).toBe();
})

});