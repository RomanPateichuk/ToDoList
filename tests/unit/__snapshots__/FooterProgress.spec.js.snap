// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента FooterProgress: соответствие снимку 1`] = `
<div>
  2/2 left
</div>
`;
